### PR TITLE
Article about the size increase of binaries when using Mandrel 23.1

### PR DESCRIPTION
**Pull request by zakkak**
Follow up to https://github.com/quarkusio/quarkusio.github.io/pull/1818

The article was written in a Jupyter notebook and exported to markdown using:

```
jupyter nbconvert index.ipynb --to markdown --NbConvertApp.output_files_dir=.
```

I tried converting it to asciidoc with:

```
jupyter nbconvert index.ipynb --to asciidoc --NbConvertApp.output_files_dir=.
```

but the output was not good. Same with `pandoc` conversion using:

```
pandoc -t asciidoc -f markdown index.md -o _posts/index.adoc
```

so I leave it as markdown.

### CODE REVIEW

File: _posts/2023-11-08-mandrel-23-1-image-size-increase.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+Both of these changes are necessary to improve the accuracy of the [async-profiler](https://github.com/async-profiler/async-profiler).
+
+## Conclusion
+
+Similarly to when Quarkus upgraded from 22.3 to 23.0, we observe an increase in the size of the generated native executables.
+Once more the changes resulting to that increase in the binary size appear to be well justified.
+As `native-image` becomes more mature and feature rich it seems inevitable to avoid increasing the size of the generated binaries.
+
+If you think that this kind of info should only be included when the user opts-in, please provide your feedback in https://github.com/oracle/graal/discussions/7707
```
[the remark this answers is not in the document]
Comment (zakkak) [reply _posts/2023-11-08-mandrel-23-1-image-size-increase.md]: Thanks. Should be fixed now.

File: _posts/2023-11-08-mandrel-23-1-image-size-increase.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
+---
+layout: post
+title: Exploring why native executables produced with Mandrel 23.1 are bigger than those produced with Mandrel 22.3
```
Comment (jerboaa) [inline _posts/2023-11-08-mandrel-23-1-image-size-increase.md:3]: ... bigger than those produced with Mandrel **23.0**

File: _posts/2023-11-08-mandrel-23-1-image-size-increase.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+2. New option `-H:±IndirectBranchTargetMarker` to mark indirect branch targets on AMD64 with an endbranch instruction. This is a prerequisite for future Intel CET support.
+3. Throw `MissingReflectionRegistrationError` when attempting to create a proxy class without having it registered at build-time, instead of a `VMError`.
+4. Support for `-XX:+HeapDumpOnOutOfMemoryError`.
+5. New `--parallelism` option to control how many threads are used by the build process.
+6. Simulation of class initializer: Class initializer of classes that are not marked for initialization at image build time are simulated at image build time to avoid executing them at image run time.
+7. and [more](https://github.com/oracle/graal/blob/master/substratevm/CHANGELOG.md#graalvm-for-jdk-21-internal-version-2310)
+
+However, it also brought an unwanted side effect.
+The native executables produced with Mandrel 23.1 are bigger than the ones produced with Mandrel 23.10.
```
Comment (jerboaa) [inline _posts/2023-11-08-mandrel-23-1-image-size-increase.md:25]: `s/23.10/23.0/`

File: _posts/2023-11-08-mandrel-23-1-image-size-increase.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+this way we detected the following two pull requests:
+
+1. [**Add support for profiling of topmost frame**](https://github.com/oracle/graal/pull/7003) which adds ~1MB of data to the image.
+2. [**ProfilingSampler does not need local variable values**](https://github.com/oracle/graal/pull/6763) (specifically [the commit "Always store bci in frame info"](https://github.com/oracle/graal/commit/d747c30c7691012c39989a8597fd850c68b740ad)) which adds ~1.7MB of data to the image.
+
+Both of these changes are necessary to improve the accuracy of the [async-profiler](https://github.com/async-profiler/async-profiler).
+
+## Conclusion
+
+Similarly to when Quarkus upgraded from 22.3 to 23.0, we observe an increase in the size of the generated native executables.
```
Comment (jerboaa) [inline _posts/2023-11-08-mandrel-23-1-image-size-increase.md:310]: Suggestion: 

```
... we observe an increase in size of the generated native executables when going from 23.0 to 23.1.
```

Comment (zakkak) [reply _posts/2023-11-08-mandrel-23-1-image-size-increase.md]: Done

**Comment by zakkak** [comment]
Thanks you @jerboaa. I addressed your comments.

**Review by jerboaa — APPROVED** [review]

**Review by gsmet — APPROVED** [review]

**Pull request merged**